base/server/python/pki/server/subsystem.py

```diff
@@ -371,12 +371,15 @@ def is_ready(self, secure_connection=True, timeout=None):
             protocol = 'http'
             port = server_config.get_unsecure_port()
 
+        # When waiting for a connection to come alive, don't bother verifying
+        # the certificate at this stage.
```

Comment:
We have to pass a bit more information into this method in all places it is called from. I think we could do that, but it seems like a lot more work than adding two lines is worth. Plus, I think we run into the same problem we do about `nssdb` above.

Reply:
Concretely, the issue with this is:

 - During [`pkispawn`](https://github.com/dogtagpki/pki/blob/master/base/server/python/pki/server/pkispawn.py#L395-L396) we try and contact the security domain.
 - We also attempt to validate the [Security Domain from config values earlier](https://github.com/dogtagpki/pki/blob/master/base/server/python/pki/server/pkispawn.py#L544).
 - However, we don't actually parse and [install the clone's certificate until later](https://github.com/dogtagpki/pki/blob/master/base/server/python/pki/server/pkispawn.py#L560-L569).

This is only a problem on a system trying to spawn an non-CA subsystem while lacking a CA. If there's already a CA on the system (primary or clone), it doesn't matter because that already has deployed the cert and it finds it successfully (see: CI).

Reply:
Yeah, we may need to make it possible for people to create the NSS database manually and import the certs/keys that they need prior to running pkispawn. If the NSS database doesn't exist yet, pkispawn will need to create it and import the PKCS #12 like it currently does, but much earlier.

Alternatively, maybe `PKIConnection` class can interactively ask the user whether to trust the incoming SSL server cert, just like the way `pki` CLI works now (and browsers too) when it receives an untrusted cert. Then the client can download and trust the CA cert from the server.

Reply:
Hmm.. I don't see `pki_ca_cert_path` in the code. Did you mean `pki_ca_signing_cert_path`?